### ISSUE TITLE
tns test from a fresh project creation doesn't work

### ISSUE THREAD

**Issue by Fedelaus** [NativeScript/nativescript-cli#4119]
**Environment**
 - CLI: 5.0.0
 - Cross-platform modules: 5.0.2 (have also tested with 5.0.0, unsure if these versions need to match with CLI)
 - Android Runtime: 5.0.0
 - iOS Runtime: 5.0.0
 - Plugin(s): 

```
Successfully prepared plugin nativescript-angular for android.
Successfully prepared plugin nativescript-theme-core for android.
Successfully prepared plugin nativescript-unit-test-runner for android.
Successfully prepared plugin tns-core-modules for android.
Successfully prepared plugin nativescript-intl for android.
Successfully prepared plugin tns-core-modules-widgets for android.
```

**Describe the bug**
When initializing tests for a fresh installation on V5, they immediately do not work and just don't even start.

**To Reproduce**
1. `tns create <appName> --ng`
2. `cd <appName> && tns test init` (I picked Jasmine for the test init)
3. `tns test android`

The tests don't run.

Output here: https://hastebin.com/oluvahequl

You can see I edit the `example.js` file, in hopes a trigger of compilation could trigger tests. 

Meanwhile my android device is reporting `No reachable hosts`. 

**Expected behavior**

The tests to run.

**Comment by Fedelaus** [comment]
Have managed to get the tests to at least execute by switching `singleRun` from false to true as per this issue here:

**Comment by Fatme** [comment]
@Fedelaus,

You can find more information here https://github.com/NativeScript/nativescript-cli/issues/4071.
Closing as a duplicate to https://github.com/NativeScript/nativescript-cli/issues/4071.

**Comment by Fedelaus** [comment]
@Fatme 

The only relation I see in that issue is that they're also trying to run `tns test`.

Their unexpected behaviour is different (their tests run immediately, mine do not).
Their android test returns socket undefined (mine does not).

I am reporting that the default, out of the box installation, does not work with tests as per the documentation.

**Comment by poppahorse** [comment]
@Fatme I too think this is unrelated to #4071  :)

This is completely out of the box v5 behaviour, and a fairly big impediment in my team using NS for serious enterprise applications.

Are there any additional reproduction steps / information we can provide? We are getting the exact same issue as @Fedelaus

**Comment by Fatme** [comment]
@Fedelaus, @poppahorse,

Sorry guys, my bad!

I don't notice the following: "No reachable hosts."

It seems your android emulators are with android 28. If this is the case you need to add `android:usesCleartextTraffic=“true”` in  `App_Resources/android/AndroidManifest.xm`.
```
<application
         android:usesCleartextTraffic=“true”
```

**Comment by irwansyahwii** [comment]
@Fatme thank you so much, your solutions save my headache

**Comment by groksrc** [comment]
I came to this issue while setting up a CI build. I also found this issue helpful, they seem to be related: https://github.com/NativeScript/nativescript-cli/issues/2017

**Comment by Fedelaus** [comment]
Thanks for the reply @Fatme.

You are correct my emu for that specific output was 28 but I was seeing the same from a default installation using 24, and I managed to get them to execute as per here: https://github.com/NativeScript/nativescript-cli/issues/4119#issuecomment-437327199

but then immediately see that output when utilizing typescript files about exports, even though the documentation just says I need to import reflect-metadata and install jasmine types.

It definitely feels like the out of the box setup requires tweaking if the documentation is actually correct.

**Comment by trylovetom** [comment]
@Fatme Why!? Why need to add this workround.

**Comment by Fatme** [comment]
@trylovetom 

This workaround was needed because the default value of `usesCleartextTraffic` was changed as of Android API 28.
According to the [docs](https://developer.android.com/guide/topics/manifest/application-element):
```
The default value for apps that target API level 27 or lower is "true". Apps that target API level 28 or higher default to "false".
```

As the default value was changed to `false`, `tns test` command didn't work out of the box as the application makes http requests runtime in order to connect to karma server. So, the user needs manually to change the value of `usesCleartextTraffic` in `AndroidManifest.xml`.

However, this workaround is no more needed as we fixed it directly in `nativescript-unit-test-runner@0.5.0`. So, if you use `nativescript-unit-test-runner@0.5.0` or higher, it should work out of the box.
